### PR TITLE
[ #5134 ] remove agda-licenses

### LINKED ISSUE #5134: Is agda-licenses obsolete?
The `agda-licences` tool requires Cabal-2.0 (limiting to ghc <= 8.6 it seems) to build and needs Agda to be build with Cabal-2.0, too, to run:
```
$ dist-newstyle/build/x86_64-osx/ghc-8.6.4/agda-licences-0.1/x/agda-licences/build/agda-licences/agda-licences ../../dist-2.6.2-ghc-8.10.3-no-code/
agda-licences: Saved package config file is outdated:
• the Cabal version changed from Cabal-3.2.1.0 to Cabal-2.0.1.1
• the compiler changed from ghc-8.10 to ghc-8.6
Re-run the 'configure' command.
```

I recently added a economic way to get the licenses using the `cabal-plan` tool, see [doc/license-report/README](https://github.com/agda/agda/blob/d6b08db36d68c87b97d37147b0b7a100ae62a073/doc/license-report/README.md).

Could we retire `agda-licences` from the codebase?  (Note that it is still picked up by e.g. `stack init`.)
Ping @nad

### PR DESCRIPTION
**Pull request by andreasabel**
Fix #5134, removing src/agda-licenses.  (See discussion there.)

### CODE REVIEW

**Comment by nad** [comment]
Some things that could be mentioned in `README.md`:

* I guess you should run `cabal v2-configure` with suitable flags (like `--enable-tests` or `-fenable-cluster-counting`).

* Agda relies on things that are not listed, including the GHC run-time system and zlib.

**Comment by andreasabel** [comment]
>     * I guess you should run `cabal v2-configure` with suitable flags (like `--enable-tests` or `-fenable-cluster-counting`).

Good point!  I also added `make configure` with a suitable invocation of `cabal configure` (did I add all necessary flags?).

>     * Agda relies on things that are not listed, including the GHC run-time system and zlib.

I see `zlib` listed in the _direct dependencies_ (just before the _indirect transitive dependencies_).

> including the GHC run-time system 

Wouldn't have thought that this has a license. (It is code produced by the compiler from the (almost) empty source file.)
Which license does it have?

**Comment by nad** [comment]
> I see `zlib` listed in the _direct dependencies_ (just before the _indirect transitive dependencies_).

That is presumably the Haskell library called `zlib`, not the [underlying library](https://zlib.net/). As another example we make use of the [ICU](http://site.icu-project.org/) library.

> > including the GHC run-time system
> 
> Wouldn't have thought that this has a license. (It is code produced by the compiler from the (almost) empty source file.)
> Which license does it have?

I don't know.

**Comment by andreasabel** [comment]
> That is presumably the Haskell library called `zlib`, not the [underlying library](https://zlib.net/). As another example we make use of the [ICU](http://site.icu-project.org/) library.

These are not captured by `cabal-plan`.

So how to proceed?  Do you approve this PR?  If not, do you want to update `agda-licenses` to work with the latest `Cabal`?

**Comment by nad** [comment]
> So how to proceed? Do you approve this PR? If not, do you want to update `agda-licenses` to work with the latest `Cabal`?

The `agda-licenses` program handles these problems by warning the user:

  https://github.com/agda/agda/blob/a9e4e1c823c86ee11cfa70b8dbd5160aa3566000/src/agda-licences/Main.hs#L29-L36

Above I suggested that you could add some text to `README.md`.

**Comment by andreasabel** [comment]
I added the disclaimer to the README.

Thanks @nad, for the comments!  I am going ahead now and merge this.

File: doc/license-report/README.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ Agda dependencies license report
 The [Makefile](Makefile) here helps you to generate a listing of the
 licenses of the Agda dependencies using `cabal-plan`.
 
+It prints the declared licence of every Cabal package that the Agda
+binary depends on (as determined by the configuration in the Cabal
+build directory `dist-newstyle`).
+Note that the binary may depend on non-Cabal packages as well.
+
+The licence printed for the rts package is that of (some version of)
+the ghc package.
```
Comment (nad) [inline doc/license-report/README.md:13]: I seem to recall that the `rts` package is not mentioned in the output of `cabal-plan`.

Comment (andreasabel) [reply doc/license-report/README.md:13]: Fixed in 8c01e3aef641accc59ae822edd4b7a79cc2b765d

**Pull request merged**